components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -439,12 +443,28 @@ private OAuthConsumerAppDTO getApplicationById(String clientId, boolean isApplic
     }
 
     private Application createOAuthApplication(ApplicationRegistrationRequest registrationRequest)
-            throws DCRMException {
+            throws DCRMException, UserStoreException {
 
         String applicationOwner = StringUtils.isNotBlank(registrationRequest.getExtApplicationOwner()) ?
                 registrationRequest.getExtApplicationOwner() :
                 PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+
         String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        /*
```

Comment:
/*
 * ApplicationOwner will be null and a server error is thrown when creating an app, if the api authentication/
 * api security is disabled for DCR endpoint.
 * In such cases, we set the tenant admin as the owner of the application.
 */